### DISCUSSION TITLE
How to hide the "notfound" template

### DISCUSSION THREAD

**Discussion by docdunning** [pinecone-router/router#31] [Q&A] [answered]
I have a page with a number of templates on it, each with an x-route attribute.  When the page loads, all these templates are hidden.  Let's say that's OK, for now, although I have started another discussion on that topic.

The problem is that this page also has a template for the 'notfound' route, and this template is displayed when the page is loaded.

```
  <template x-route="notfound">
    <p>Oops - cannot find that route.</p>
  </template>
```

I obviously don't want to see this message when the page loads.  How can I hide it but still have it work if a bad route is used?

**Comment by rehhouari** [comment]
There must be an issue can you provide a reproduction project?

The template content should not show until the specified route is matched, i can't tell why that's happening without a reproduction project
have you tried using [`x-cloak`](https://alpinejs.dev/directives/cloak)?

**Comment by docdunning** [comment, marked as answer]
OK - this was down to my misunderstanding of the way that routes are matched.  See discussion #30
